webapp/src/services/api/studies/raw/index.ts

```diff
@@ -45,9 +85,54 @@ export async function importFile(params: ImportFileParams) {
   });
 }
 
-export async function deleteFile(params: DeleteFileParams) {
+/**
+ * Deletes a raw file from a study.
+ *
+ * @param params - Parameters for deleting the file
+ * @param params.studyId - Unique identifier of the study
+ * @param params.path - Path to the file to delete
+ * @returns Promise that resolves when the deletion is complete
+ */
+export async function deleteFile(params: DeleteFileParams): Promise<void> {
   const { studyId, path } = params;
-  const url = `/v1/studies/${studyId}/raw`;
+  await client.delete(`/v1/studies/${studyId}/raw`, { params: { path } });
+}
+
+/**
+ * Reads an original raw file from a study with its metadata.
+ *
+ * @param studyId - Unique identifier of the study
+ * @param filePath - Path to the file within the study
+ * @returns Promise containing the file data and metadata
+ */
+export async function getRawFile(
+  studyId: string,
+  filePath: string,
+): Promise<RawFile> {
+  const { data, headers } = await client.get<RawFile["data"]>(
+    `/v1/studies/${studyId}/raw/original-file`,
+    {
+      params: {
+        path: filePath,
+      },
+      responseType: "blob",
+    },
+  );
+
+  // Get the original file name from the response Headers
+  const contentDisposition = headers["content-disposition"];
+  let filename = filePath.split("/").pop() || "file"; // fallback filename
+
+  if (contentDisposition) {
+    const matches = /filename=([^;]+)/.exec(contentDisposition);
+
+    if (matches?.[1]) {
+      filename = matches[1].replace(/"/g, "").trim();
+    }
+  }
 
-  await client.delete<void>(url, { params: { path } });
+  return {
+    data,
+    filename,
```

Comment:
the filename extraction from content-disposition is indeed needed

- The filepath indicates the storage location it can differ from the original filename
- It's consistent with file download best practices where the original filename comes from content-disposition
- This maintains a single source of truth (depending on two distinct sources of truth, is a bad practice) if the filepath changes and does not reflect the original name anymore => bug

Reply:
ok will try this